src/encoded/static/components/experiment.js

```diff
@@ -302,6 +302,46 @@ LibrarySubmitterComments.propTypes = {
 };
 
 
+/**
+ * Display, as one element of a "key-value" definition list, all the `construction_platform`
+ * properties in a dataset's replicates' library.
+ */
+const DatasetConstructionPlatform = ({ context }) => {
+    // Collect the library construction platforms from all the dataset replicates.
+    const constructionPlatforms = context.replicates.reduce((platforms, replicate) => {
+        // We easily have duplicates, so only add the replicate library platform if we haven't
+        // already seen it.
+        if ((replicate.library && replicate.library.construction_platform) &&
+            (platforms.findIndex(platform => platform['@id'] === replicate.library.construction_platform['@id']) === -1)) {
```

Comment:
Is it a good idea to put the "platforms.findIndex(platform => platform['@id'] === replicate.library.construction_platform['@id'])" onto its own line. It is the last parenthesis can easily be missed and it easy to think you are doing something like x===y===-1.